lib/Data/Resample.pm

```diff
@@ -0,0 +1,355 @@
+package Data::Resample;
+
+use strict;
+use warnings;
+
+use 5.010;
+use Moose;
+
+use RedisDB;
+use Date::Utility;
+use Sereal::Encoder;
+use Sereal::Decoder;
+
+use MooseX::Types::Moose qw(Int Num Str);
+use MooseX::Types -declare => [qw(
+        interval
+        )];
+
+use Moose::Util::TypeConstraints;
+use Time::Duration::Concise;
+
+subtype 'interval', as 'Time::Duration::Concise';
+coerce 'interval', from 'Str', via { Time::Duration::Concise->new(interval => $_) };
+
+=head1 NAME
+
+Data::Resample - A module that allows to resample a data feed. 
+
+=head1 VERSION
+
+Version 0.01
+
+=head1 SYNOPSIS
+
+  use Data::Resample::TicksCache;
+  use Data::Resample::ResampleCache;
+
+  my $ticks_cache = Data::Resample::TicksCache->new({
+        redis => $redis,
+        });
+
+  my %tick = (
+        symbol => 'USDJPY',
+        epoch  => time,
+        quote  => 103.0,
+        bid    => 103.0,
+        ask    => 103.0,
+  );
+
+  $ticks_cache->tick_cache_insert(\%tick);
+
+  my $ticks = $ticks_cache->tick_cache_get_num_ticks({
+        symbol => 'USDJPY',
+        });
+
+  my $resample_cache = Data::Resample::ResampleCache->new({
+        redis => $redis,
+        });
+
+=head1 DESCRIPTION
+
+A module that allows you to resample a data feed
+
+=cut
+
+our $VERSION = '0.01';
+
+=head1 ATTRIBUTES
+=cut
+
+=head2 sampling_frequency
+
+=head2 tick_cache_size
+
+=head2 resample_cache_size
+
+=cut
+
+has sampling_frequency => (
+    is      => 'ro',
+    isa     => 'interval',
+    default => '15s',
+    coerce  => 1,
+);
+
+has tick_cache_size => (
+    is      => 'ro',
+    default => 1860,
+);
+
+has resample_cache_size => (
+    is      => 'ro',
+    default => 2880,
+);
+
+has agg_retention_interval => (
+    is      => 'ro',
+    isa     => 'interval',
+    lazy    => 1,
+    coerce  => 1,
+    builder => '_build_agg_retention_interval',
+);
+
+sub _build_agg_retention_interval {
```

Comment:
maybe change the name to '_build_resample_retention_interval'. We should remove all references like 'agg', 'unagg' etc from the code.